src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2881,6 +2888,176 @@ public ConstructorEdit(INamedTypeSymbol oldType)
             return semanticEdits.ToImmutable();
         }
 
+        private void TryReportFieldAttributeRudeEdits(ArrayBuilder<RudeEditDiagnostic> diagnostics, EditAndContinueCapabilities capabilities, SemanticModel? oldModel, SemanticModel newModel, Edit<SyntaxNode> edit, ArrayBuilder<SemanticEditInfo> semanticEdits, CancellationToken cancellationToken)
+        {
+            // We have to get symbols for the individual fields declared within them to check for attribute rude edits.
+            // The attributes apply to all fields in the declaration.
+            var (oldFieldSymbol, newFieldSymbol) = GetSymbolsForField(oldModel, edit.OldNode, newModel, edit.NewNode, cancellationToken);
+            if (newFieldSymbol is not null)
+            {
+                ReportAttributeEdits(oldFieldSymbol, newFieldSymbol, capabilities, diagnostics, semanticEdits, null, cancellationToken);
+            }
+        }
+
+        private void ReportAttributeEdits(ISymbol? oldSymbol, ISymbol newSymbol, EditAndContinueCapabilities capabilities, ArrayBuilder<RudeEditDiagnostic> diagnostics, ArrayBuilder<SemanticEditInfo>? semanticEdits, Func<SyntaxNode, SyntaxNode?>? syntaxMap, CancellationToken cancellationToken)
```

Comment:
>ReportAttributeEdits [](http://example.com/codeflow?start=21&length=20)

I'd call this `AnalyzeCustomAttributes` - we use `ReportXxx` for rude edit reporting not a general update analysis.